packages/arb-token-bridge-ui/src/token-bridge-sdk/CctpTransferStarter.ts

```diff
@@ -48,29 +50,33 @@ export class CctpTransferStarter extends BridgeTransferStarter {
     return allowance.lt(amount)
   }
 
-  public async approveToken({ signer, amount }: ApproveTokenProps) {
-    const sourceChainId = await getChainIdFromProvider(this.sourceChainProvider)
+  public async approveTokenPrepareTxRequest(params?: {
+    amount: BigNumber | undefined
+  }): Promise<TransactionRequest> {
+    const {
+      //
+      usdcContractAddress,
+      tokenMessengerContractAddress
+    } = getCctpContracts({ sourceChainId: await this.getSourceChainId() })
 
-    const { usdcContractAddress, tokenMessengerContractAddress } =
-      getCctpContracts({ sourceChainId })
+    return {
+      to: usdcContractAddress,
+      data: ERC20__factory.createInterface().encodeFunctionData('approve', [
+        tokenMessengerContractAddress,
+        params?.amount ?? constants.MaxInt256
```

Comment:
note that `constants.MaxInt256` is different from `constants.MaxUint256`

maximum value should be the latter because the arg is of type `uint256` according to ERC20 ABI

Reply:
this was an error from before, fixed now